### PR TITLE
Feat: Add support for Go

### PR DESCRIPTION
**Pull request by matbur**
I added yet another language support: Go.

I faced a few problems:
1. After generating examples some files changed for python, java and c. Should I split it to next PR?
2. I was unable to run generating examples without this line in script: `sys.path.append('.')`

### CODE REVIEW

**Comment by krinart** [comment]
Hey @matbur, thanks a lot for this great PR. Great job! You we are able to successfully capture all the details of the architecture.

I left some minor comments and wanted to mention couple of things:
1. Please update also `cli.py` in order to properly support go.

2. `After generating examples some files changed for python, java and c. Should I split it to next PR?`
I would ask you to verify that without your changes we get exactly the same code which is currently under the `generated_code_examples` folder. We'll see what to do with them depending on the result.

3. `I was unable to run generating examples without this line in script: sys.path.append('.')`
That's kinda by design. If you make `python setup.py develop` in the root folder you should be able to run the script without that fix.

4. I would also ask you to manually verify that you get proper results on a big model, for example `RandomForest` with at least 200-300 and maybe even 500 estimators. It may take some time but it is important to know that we support them. For example Python and Java required some additional effort to make them support such huge models (that's why we have `BinExpressionDepthTrackingMixin` and `SubroutinesAsFunctionsMixin` mixins)

Again thanks a lot for your contribution, it's amazing!

**Comment by krinart** [comment]
Oh and there are 4 `flake8` errors:
```
./m2cgen/__init__.py:1:80: E501 line too long (82 > 79 characters)
./m2cgen/interpreters/go/code_generator.py:26:80: E501 line too long (80 > 79 characters)
./m2cgen/interpreters/go/code_generator.py:44:80: E501 line too long (86 > 79 characters)
./tests/e2e/executors/go.py:37:1: W293 blank line contains whitespace
```

Please fix them to make Travis happy.

**Comment by matbur** [comment]
@krinart thanks for your code review. I decided to split updating generated code to separate PR: #80

File: m2cgen/interpreters/go/code_generator.py [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+        self.add_function_def(name, args, is_scalar_output)
+        yield
+        self.add_block_termination()
+
+    def add_var_declaration(self, size):
+        var_name = self.get_var_name()
+        tpl = self.tpl_vector_var_declare if size > 1 else self.tpl_scalar_var_declare
+
+        self.add_code_line(tpl(var_name=var_name, size=size))
+        return var_name
```
[the remark this answers is not in the document]
Comment (matbur) [reply m2cgen/interpreters/go/code_generator.py]: It's a little bit more complicated. I will remove function `_get_var_type ` and implement `_get_var_declare_type` instead

File: setup.py [hunk 2]
```diff
@@ -2,13 +2,13 @@
 
 setup(
     name="m2cgen",
-    version="0.1.2",
+    version="0.1.3",
```
Comment (izeigerman) [inline setup.py]: The `0.1.2` hasn't been released yet, so it doesn't make sense to bump the minor version. However, I think this update deserves to be `0.2.0` so why don't we make it so? :)

**Pull request merged**